src/vm/jitinterface.cpp

```diff
@@ -9953,8 +9953,8 @@ void* CEEInfo::getPInvokeUnmanagedTarget(CORINFO_METHOD_HANDLE method,
 
 /*********************************************************************/
     // return address of fixup area for late-bound N/Direct calls.
-void* CEEInfo::getAddressOfPInvokeFixup(CORINFO_METHOD_HANDLE method,
-                                        void **ppIndirection)
```

Comment:
Yes, the direct P/Invoke is only allowed when the marshalling is not required (e.g. when `pInvokeMarshalingRequired` returns false).